src/ucs/config/parser.c

```diff
@@ -1066,6 +1066,173 @@ void ucs_config_help_table(char *buf, size_t max, const void *arg)
     snprintf(buf, max, "Table");
 }
 
+/**
+ * Return number of allowed keys in the given config
+ */
+static size_t ucs_config_key_value_count(const ucs_config_key_field_t *cfg_key)
+{
+    size_t count = 0;
+    for (; cfg_key->name; ++cfg_key, ++count);
+    return count;
+}
+
+/**
+ * Given key-value pair, find appropriate destination for storing value in the
+ * temporary values buffer, and save it there
+ */
+static ucs_status_t
+ucs_config_key_value_set(const ucs_config_key_field_t *cfg_key,
+                         const char **values, size_t key_count,
+                         const char *key, const char *value)
+{
+    size_t index;
+
+    if (NULL == value) {
+        /* Empty value means that key is a generic value for all keys */
+        for (index = 0; index < key_count; ++index) {
+            values[index] = key;
+        }
+        return UCS_OK;
+    } else {
+        /* Key configured, save value for specific key */
+        for (index = 0; index < key_count; ++index, ++cfg_key) {
```

Comment:
can we reuse ucs_string_find_in_list?

Reply:
Unfortunately we can't.
Because we search in NULL terminated array of key configs (`ucs_config_key_field_t *`), not in array of strings